sprocs/check_assessment_access_rule.sql

```diff
@@ -74,12 +74,30 @@ BEGIN
             ORDER BY r.access_end DESC -- choose the longest-lasting if >1
             LIMIT 1;
 
-            IF NOT FOUND THEN
-                -- no reservation so block access
-                authorized := FALSE;
+            IF FOUND THEN
+                -- we have a valid reservation, don't keep going to "authorized := FALSE"
+                EXIT schedule_access;
+            END IF;
+
+            -- is there a checked-in pt_reservation?
+            PERFORM 1
+            FROM
+                pt_reservations AS r
+                JOIN pt_enrollments AS e ON (e.id = r.enrollment_id)
+                JOIN pt_exams AS x ON (x.id = r.exam_id)
+            WHERE
+                (date BETWEEN r.access_start AND r.access_end)
+                AND e.user_id = check_assessment_access_rule.user_id
+                AND x.uuid = assessment_access_rule.exam_uuid;
```

Comment:
Do we want the same "pick at most one, and pick the longest-lasting one if there are multiple" logic that we use for PS?

Reply:
From what I can see, that logic isn't really doing anything. It selects into the block-local variable `reservation`, but we never refer to this variable again. So I can't see that it matters which `reservation` we pick, or even if we pick one at all. This is why I just used `PERFORM 1`. Does this also make sense to you?

Reply:
Ah! That makes perfect sense. As long as _some_ reservation is active at this point, that's all we care about.

Reply:
Exactly!